src/main/java/formflow/library/data/SubmissionRepositoryService.java

```diff
@@ -85,9 +88,18 @@ public Submission findOrCreate(HttpSession httpSession) {
     var id = (UUID) httpSession.getAttribute("id");
     if (id != null) {
       Optional<Submission> submissionOptional = findById(id);
-      return submissionOptional.orElseGet(Submission::new);
+      if (submissionOptional.isEmpty()) {
+        log.error("findOrCreate could not find submission: " + id);
+        Submission newSubmission = new Submission();
+        log.info("findOrCreate created new submission: " + newSubmission.getId());
+        return newSubmission;
+      } else {
+        return submissionOptional.get();
+      }
     } else {
-      return new Submission();
+      Submission newSubmission = new Submission();
+      log.info("findOrCreate got no submission id from session, so created new submission: " + newSubmission.getId());
```

Comment:
note that the new submission id should be "null" since it hasn't been saved yet and won't get an id until it's in the DB (the DB assigns the id).

Reply:
We added another log in the `save` method to get that newly created id that show up `null` here.